core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -195,7 +196,11 @@ object ChunkedByteBuffer {
     val is = new FileInputStream(file)
     ByteStreams.skipFully(is, offset)
     val in = new LimitedInputStream(is, length)
-    val chunkSize = math.min(maxChunkSize, length).toInt
+    // Though in theory you should be able to index into an array of size Int.MaxValue, in practice
+    // jvms don't let you go up to limit.  It seems you may only need - 2, but we leave a little
+    // extra room.
+    val maxArraySize = Int.MaxValue - 512
```

Comment:
Looks like the max is safely more like `Int.MaxValue - 5` from things like https://www.quora.com/What-is-the-maximum-size-of-the-array-in-Java You could probably push it further, but whatever.

Reply:
We already had seen the similar problem in other places. Can we use [this value](https://github.com/apache/spark/blob/master/common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java#L52) here, too?

Reply:
great suggestion, thanks, just updated